### PR TITLE
[storescu] Enable sending more than one file

### PR DESCRIPTION
**Pull request by natdash**
Addresses #182

### CODE REVIEW

File: storescu/src/main.rs [hunk 1]
```diff
@@ -39,183 +41,215 @@ struct App {
...
+    for mut file in &mut dicom_files {
+        // TODO(#106) transfer syntax conversion is currently not supported
+        match check_presentation_contexts(file, scu.presentation_contexts()) {
+            Ok((pc, ts)) => {
+                file.pc_selected = Some(pc);
+                file.ts_selected = Some(ts);
+            },
+            Err(e) => {
+                if fail_first {
+                    eprintln!("Could not choose a transfer syntax: {}", e);
```
Comment (Enet4) [inline storescu/src/main.rs:131]:
```suggestion
                eprintln!("Could not choose a transfer syntax: {}", e);
                if fail_first {
```
The error should still be reported to the user.

**Review by Enet4 — COMMENTED** [review]
Thank you for working on this!

The requirement to scan the directory tree for files was included from the start in #211, because it makes the SCU much easier to work it. I would be grateful if you extended this contribution to make a recursive listing of files to send, so that the command works by passing a directory of DICOM files.

One other concern is that unless `verbose` is set, the tool is currently rather silent, giving no clue on how much work has done. Using a lib such as [`indicatif`](https://crates.io/crates/indicatif) to show the current progress should be easy to integrate and make the output much more appealing.

**Comment by natdash** [comment]
Thanks for the feedback! I'll update to address and add in the missed functionality. Adding a progress bar sounds great, I'll work on that as well.

**Comment by natdash** [comment]
I'm unsure why the CI check failed - it seems to be an error accessing a test DICOM file?

**Comment by Enet4** [comment]
> I'm unsure why the CI check failed - it seems to be an error accessing a test DICOM file?

It's not much to worry about, every once in a while there appears to be a sporadic failure in `dicom-test-files`. Might be worth diagnosing further if it happens again.

Thank you for including the requested capabilities! I'll review and test this on my end once possible.

File: storescu/src/main.rs [hunk 2]
```diff
@@ -39,183 +43,246 @@ struct App {
...
-    if verbose {
-        println!("Establishing association with '{}'...", &addr);
+        match check_file(file) {
+            Ok(dicom_file) => {
+                presentation_contexts.insert((dicom_file.sop_class_uid.to_string(),dicom_file.file_transfer_syntax.clone()));
+                dicom_files.push(dicom_file);
+            },
+            Err(e) => {
+                if verbose {
+                    println!("Error: {}", e);
```
Comment (Enet4) [inline storescu/src/main.rs:110]:
```suggestion
                    println!("Could not open file {}: {}", file.display(), e);
```

**Review by Enet4 — COMMENTED** [review]
I've been playing around and it functions pretty much as intended! For now, I have just a few more suggestions to apply before merging.

**Comment by natdash** [comment]
That's great, thanks again! I've implemented the suggested improvements now.

**Review by Enet4 — APPROVED** [review]
Wonderful! In the meantime, I discovered some issues in the ul crate which also affect the store SCU (#215), but I'm already working on it.

Thank you for your contributions.

**Pull request merged**